### PR TITLE
Don't generate docs when "quickly" sys prop is set to true

### PR DESCRIPTION
**Pull request by geoand**
Follows up on: #10230

### CODE REVIEW

File: core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java [hunk 1]
```diff
@@ -67,7 +67,8 @@
             SYSTEM_PROPERTIES.getProperty("maven.multiModuleProjectDirectory", "."));
     public static final Path GENERATED_DOCS_PATH = Paths.get(DOCS_OUT_DIR + DOCS_SRC_MAIN_ASCIIDOC_GENERATED).toAbsolutePath();
     public static final File GENERATED_DOCS_DIR = GENERATED_DOCS_PATH.toFile();
-    public static final Boolean SKIP_DOCS_GENERATION = Boolean.valueOf(SYSTEM_PROPERTIES.getProperty("skipDocs", "false"));
+    public static final Boolean SKIP_DOCS_GENERATION = Boolean.parseBoolean(SYSTEM_PROPERTIES.getProperty("skipDocs", "false"))
+            || Boolean.parseBoolean(SYSTEM_PROPERTIES.getProperty("quickly", "false"));
```
Comment (Sanne) [inline core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java:71]: should be `"true"` ?

**Review by Sanne — CHANGES_REQUESTED** [review]

File: core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java (hunk 1 above)
Comment (Sanne) [reply core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java:71]: ah, sorry that's the default.

Comment (geoand) [reply core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java:71]: Currently skipping the documentation generation is opt-in, so we don't want to skip by default

**Review by Sanne — APPROVED** [review]

File: core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java (hunk 1 above)
Comment (Sanne) [reply core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java:71]: yes, sure I got confused by the API :)

Comment (geoand) [reply core/processor/src/main/java/io/quarkus/annotation/processor/Constants.java:71]: :)

**Review by machi1990 — APPROVED** [review]
LGTM, thanks for the followup!

**Comment by geoand** [comment]
Thanks for the review folks!

**Comment by gsmet** [comment]
I'm a bit torn about that one.

The reason is that it's nice to be able to have a fast build with the documentation generated as it's the only way to get a clean doc generation without inclusion warnings hiding other potential issues.

So I wonder if I would keep it the way it is and let people add `-DskipDocs` if they want to disable it.

**Comment by geoand** [comment]
IMHO, having to specify `-DskipDocs` when doing `-Dquickly`, kind of defeats the purpose of setting the `skipDocs` property in the `quickly` profile.
Moreover, I am pretty sure that the most common usage of `quickly` is to build the code, not the docs.

**Comment by gsmet** [comment]
Well the thing is that you also need to check the doc at some point.

If we are sure it won't end up with having a ton of PRs with doc issues then go ahead.

**Comment by geoand** [comment]
I'm obviously not sure, I just think that this caters for the most common use case

**Comment by geoand** [comment]
So I'll leave it up to you :)

**Comment by geoand** [comment]
Looked at this again this morning and I think it's very counter intuitive to have to add `-DskipDocs` to `-Dquickly`. 
For me, when one needs to create the docs, they simply shouldn't use `-Dquickly` (which set the `skipDocs` property anyway)...

**Comment by gsmet** [comment]
@geoand so what should they do? Because when checking the docs properly, you need to build the whole tree.

Anyway, I have my own aliases now as I don't think I will agree with all of you so feel free to do whatever you want with quickly :).

**Comment by geoand** [comment]
The way I see quicky being used, is in cases where code needs to be built, not docs.

What I would suggest for the docs, is exactly what you ended up doing, having your own aliases :)

**Comment by gsmet** [comment]
@geoand either you merge it or I close it, your call! :)

**Comment by geoand** [comment]
I am in favor of merging, but CI seems to be stuck?

**Comment by geoand** [comment]
I just rebased, let's see what happens

**Pull request merged**